tests/unit/fem/test_transfer.cpp

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "catch.hpp"
+#include "mfem.hpp"
+
+using namespace mfem;
+
+int dimension;
+double coeff(const Vector& x)
+{
+   if (dimension == 2)
+   {
+      return 1.1 * x[0] + 2.0 * x[1];
+   }
+   else
+   {
+      return 1.1 * x[0] + 2.0 * x[1] + 3.0 * x[2];
+   }
+}
+
+void vectorcoeff(const Vector& x, Vector& y)
+{
+   y(0) = coeff(x);
+   y(1) = -coeff(x);
+   if (dimension == 3)
+   {
+      y(2) = 2.0 * coeff(x);
+   }
+}
+
+
+TEST_CASE("transfer")
+{
+   for (int vectorspace = 0; vectorspace <= 1; ++vectorspace)
+   {
+      for (dimension = 2; dimension <= 3; ++dimension)
+      {
+         for (int ne = 1; ne <= 3; ++ne)
+         {
+            for (int order = 1; order <= 4; order *= 2)
+            {
+               for (int geometric = 0; geometric <= 1; ++geometric)
+               {
+
+                  int fineOrder = (geometric == 1) ? order : 2 * order;
+
+                  std::cout << "Testing transfer:\n"
+                            << "  Vectorspace:  " << vectorspace << "\n"
+                            << "  Dimension:    " << dimension << "\n"
+                            << "  Elements:     " << std::pow(ne, dimension) << "\n"
+                            << "  Coarse order: " << order << "\n"
+                            << "  Fine order:   " << fineOrder << "\n"
+                            << "  Geometric:    " << geometric << "\n";
+
+                  Mesh* mesh;
+                  if (dimension == 2)
+                  {
+                     mesh = new Mesh(ne, ne, Element::QUADRILATERAL, 1, 1.0, 1.0);
```

Comment:
I'm probably nitpicking, but would it be easy to add a test for tris/tets? The p-refinement in particular has a different code path in this case that I don't think is unit tested at the moment.

Reply:
It was quite easy to add. Triangles and tetrahedra are being tested now too.